### PR TITLE
Fix estimation of number of tests per repeated test run

### PR DESCRIPTION
**Pull request by dhiller**
**What this PR does / why we need it**:
Due to grep returning non zero exit code in case of no match we rewrite
the grepping logic to ignore these errors. This also makes it a bit
better to read as the awk sum is replaced by proper arithmetic
expressions.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
@oshoval at some time in the past there was a convention of using `_test.go` for test files. As this is no more valid I need to adjust the logic to reflect that.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by oshoval** [comment]
Thanks Daniel,
from what i know we removed (Edy) _test from some network files,
but it doesnt mean that other files don't have _test still

will this PR handle both cases?

@EdDev FYI

another point, is it fine that the check for flakes lane is't mandatory ?

**Comment by dhiller** [comment]
> Thanks Daniel,
> from what i know we removed (Edy) _test from some network files,
> but it doesnt mean that other files don't have _test still
> 
> will this PR handle both cases?

Yes now it greps all files for test methods, ignoring the errors that grep generates in case no test method has been found.

> Just making sure
> 
> @EdDev FYI
> 
> another point, is it fine that the check for flakes lane is't mandatory ?

Yes, as it should only make visible flaky tests, but not block on them. I think with the quarantine solution we now have in place we are good. 

**Note:**  I think this only is a short term solution and in the long term the tests binary should support only running changed tests through a flag. See #5175

File: automation/repeated_test.sh [hunk 1]
```diff
@@ -44,6 +44,28 @@ function new_tests {
...
 
+# taking into account that each file containing changed tests is run several times per default
+# and considering overhead of cluster-up etc., we should just skip the run
+# if the total number of tests for all runs gets higher than the total number of tests
+function should_skip_test_run_due_to_too_many_tests() {
+    local new_tests="$1"
+    local test_start_pattern='(Specify|It|Entry)\('
+    local tests_total_estimate=0
+    while IFS= read -r -d '' test_file_name; do
+        tests_total_estimate=$(( tests_total_estimate + $(grep -hcE "${test_start_pattern}" "$test_file_name") ))
```
Comment (oshoval) [inline automation/repeated_test.sh:55]: $ find tests/ -name '*.go'  | xargs grep It | wc -l
765
$ find tests/ -name '*.go'  | xargs grep "It(" | wc -l 
513 

Please consider adding "(" so the count will be more accurate
we can see it affects

**Review by oshoval — APPROVED** [review]
Thanks

**Review by davidvossel — APPROVED** [review]
/approve

**Pull request merged**